client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -604,10 +517,11 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 		}
 
 		const selectStyle = () => {
-			if (
-				shouldLimitGlobalStyles &&
-				! isDefaultGlobalStylesVariationSlug( selectedStyleVariation?.slug )
-			) {
+			const hasSelectedGlobalStyles =
+				! isDefaultGlobalStylesVariationSlug( selectedStyleVariation?.slug ) ||
+				!! selectedFontVariation;
+
+			if ( shouldLimitGlobalStyles && hasSelectedGlobalStyles ) {
 				unlockPremiumGlobalStyles();
```

Comment:
I think we need to modify this function so that it will show the modal when a premium font is selected in a free site? Otherwise, the `Continue` button will do nothing and the flow cannot progress.

Reply:
Oops...I only tested on style variations 🤦‍♂️